src/lib/parts.ts

```diff
@@ -321,7 +321,7 @@ export class NodePart implements Part {
     if (partIndex < itemParts.length) {
       // Truncate the parts array so _value reflects the current state
       itemParts.length = partIndex;
-      this.clear(itemPart && itemPart!.endNode);
+      this.clear(itemPart && itemPart.endNode);
```

Comment:
Wouldn't `this.clear(itemPart!.endNode)` also work? I'm not super familiar with TypeScript so it might be functionally different, but it seems more in line with general TypeScript style.

Reply:
The `!` has no runtime effect, it's the programmer saying to the type checker "I know you can't tell, but I can, and that expression can never be null or undefined here"

Reply:
Ah, I see. I thought it functioned like a Safe Navigation Operator. Thanks :)